### PR TITLE
fix update bpfapplication object

### PR DESCRIPTION
**Pull request by msherif1234**
**- What this PR does and why is it needed**
The following e2e test case was failing because bpfapplication wasn't updated correctly now with the fix it pass
```shell
WHAT="non existent interface name in unrelated IngressNodeFirewall doesn't block application of new IngressNodeFirewalls policies" make test-e2e 
/home/mmahmoud/go/src/ingress-node-firewall/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
==== Generating DaemonSet manifest
hack/generate-daemon-manifest.sh
/home/mmahmoud/go/src/ingress-node-firewall/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
cp bundle/manifests/* manifests/stable
rm -rf /tmp/test_validation_logs/
mkdir -p /tmp/test_validation_logs/
go test --tags=validationtests -v ./test/e2e/validation -ginkgo.v -junit /tmp/test_validation_logs/ -report /tmp/test_validation_logs/ -ginkgo.focus "non existent interface name in unrelated IngressNodeFirewall doesn't block application of new IngressNodeFirewalls policies"
=== RUN   TestValidation
Running Suite: Ingress Node Firewall Operator Validation Suite
==============================================================
Random Seed: 1722455142
Will run 0 of 4 specs

SSSS
JUnit report was created: /tmp/test_validation_logs/validation_junit.xml

Ran 0 of 4 Specs in 0.000 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 4 Skipped

You're using deprecated Ginkgo functionality:
=============================================
Ginkgo 2.0 is under active development and will introduce several new features, improvements, and a small handful of breaking changes.
A release candidate for 2.0 is now available and 2.0 should GA in Fall 2021.  Please give the RC a try and send us feedback!
  - To learn more, view the migration guide at https://github.com/onsi/ginkgo/blob/ver2/docs/MIGRATING_TO_V2.md
  - For instructions on using the Release Candidate visit https://github.com/onsi/ginkgo/blob/ver2/docs/MIGRATING_TO_V2.md#using-the-beta
  - To comment, chime in at https://github.com/onsi/ginkgo/issues/711

  You are using a custom reporter.  Support for custom reporters will likely be removed in V2.  Most users were using them to generate junit or teamcity reports and this functionality will be merged into the core reporter.  In addition, Ginkgo 2.0 will support emitting a JSON-formatted report that users can then manipulate to generate custom reports.

  If this change will be impactful to you please leave a comment on https://github.com/onsi/ginkgo/issues/711
  Learn more at: https://github.com/onsi/ginkgo/blob/ver2/docs/MIGRATING_TO_V2.md#removed-custom-reporters

To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=1.16.5

--- PASS: TestValidation (0.00s)
PASS
ok  	github.com/openshift/ingress-node-firewall/test/e2e/validation	0.020s
rm -rf /tmp/test_e2e_logs/
mkdir -p /tmp/test_e2e_logs/
go test -timeout 20m --tags=e2etests -v ./test/e2e/functional -ginkgo.v -junit /tmp/test_e2e_logs/ -report /tmp/test_e2e_logs/ -ginkgo.focus "non existent interface name in unrelated IngressNodeFirewall doesn't block application of new IngressNodeFirewalls policies"
=== RUN   TestE2E
Running Suite: Ingress Node Firewall Operator E2E Suite
=======================================================
Random Seed: 1722455143
Will run 1 of 22 specs

SSSSSSSSSS
------------------------------
Ingress Node Firewall IngressNodeFirewall 
  non existent interface name in unrelated IngressNodeFirewall doesn't block application of new IngressNodeFirewalls policies
  /home/mmahmoud/go/src/ingress-node-firewall/test/e2e/functional/tests/e2e.go:856
STEP: [IPV4] Initial connectivity check for protocol TCP from pod "e2e-inf-client-one" to destination pod "e2e-inf-server-one"
STEP: Creating Ingress node firewall rules &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:e2e-inf-0 GenerateName: Namespace: SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[e2e-inf-test:] Annotations:map[] OwnerReferences:[] Finalizers:[] ManagedFields:[]} Spec:{NodeSelector:{MatchLabels:map[node-role.kubernetes.io/worker:] MatchExpressions:[]} Ingress:[{SourceCIDRs:[10.244.2.57/32] FirewallProtocolRules:[{Order:1 ProtocolConfig:{Protocol:TCP TCP:0xc000b0a960 UDP:<nil> SCTP:<nil> ICMP:<nil> ICMPv6:<nil>} Action:Deny}]}] Interfaces:[doesntexist]} Status:{SyncStatus:}}
STEP: Creating Ingress node firewall rules &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:e2e-inf-1 GenerateName: Namespace: SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[e2e-inf-test:] Annotations:map[] OwnerReferences:[] Finalizers:[] ManagedFields:[]} Spec:{NodeSelector:{MatchLabels:map[node-role.kubernetes.io/worker:] MatchExpressions:[]} Ingress:[{SourceCIDRs:[10.244.2.57/32] FirewallProtocolRules:[{Order:2 ProtocolConfig:{Protocol:TCP TCP:0xc000b0a980 UDP:<nil> SCTP:<nil> ICMP:<nil> ICMPv6:<nil>} Action:Deny}]}] Interfaces:[eth0]} Status:{SyncStatus:}}
STEP: [IPV4] Check connectivity after applying IngressNodeFirewall rules for protocol TCP from pod "e2e-inf-client-one"/10.244.2.57 to destination pod "e2e-inf-server-one"/10.244.1.66
STEP: [IPV4] Checking if drop events created

• [SLOW TEST:23.713 seconds]
Ingress Node Firewall
/home/mmahmoud/go/src/ingress-node-firewall/test/e2e/functional/tests/e2e.go:123
  IngressNodeFirewall
  /home/mmahmoud/go/src/ingress-node-firewall/test/e2e/functional/tests/e2e.go:132
    non existent interface name in unrelated IngressNodeFirewall doesn't block application of new IngressNodeFirewalls policies
    /home/mmahmoud/go/src/ingress-node-firewall/test/e2e/functional/tests/e2e.go:856
------------------------------
SSSSSSSSSSS
JUnit report was created: /tmp/test_e2e_logs/e2e_junit.xml

Ran 1 of 22 Specs in 23.723 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 21 Skipped

You're using deprecated Ginkgo functionality:
=============================================
Ginkgo 2.0 is under active development and will introduce several new features, improvements, and a small handful of breaking changes.
A release candidate for 2.0 is now available and 2.0 should GA in Fall 2021.  Please give the RC a try and send us feedback!
  - To learn more, view the migration guide at https://github.com/onsi/ginkgo/blob/ver2/docs/MIGRATING_TO_V2.md
  - For instructions on using the Release Candidate visit https://github.com/onsi/ginkgo/blob/ver2/docs/MIGRATING_TO_V2.md#using-the-beta
  - To comment, chime in at https://github.com/onsi/ginkgo/issues/711

  You are using a custom reporter.  Support for custom reporters will likely be removed in V2.  Most users were using them to generate junit or teamcity reports and this functionality will be merged into the core reporter.  In addition, Ginkgo 2.0 will support emitting a JSON-formatted report that users can then manipulate to generate custom reports.

  If this change will be impactful to you please leave a comment on https://github.com/onsi/ginkgo/issues/711
  Learn more at: https://github.com/onsi/ginkgo/blob/ver2/docs/MIGRATING_TO_V2.md#removed-custom-reporters

To silence deprecations that can be silenced set the following environment variable:
  ACK_GINKGO_DEPRECATIONS=1.16.5

--- PASS: TestE2E (23.73s)
PASS
ok  	github.com/openshift/ingress-node-firewall/test/e2e/functional	23.769s
```

### CODE REVIEW

**Review by anfredette — COMMENTED** [review]
I noticed a typo in the existing code:  BpfApplicatopn -> BpfApplication

**Comment by anfredette** [comment]
Also, changing ImagePullPolicy: bpfmaniov1alpha1.PullAlways to PullIfNotPresent may speed things up after the first time the image is downloaded.

File: pkg/bpf-mgr/lib-bpfman.go [hunk 1]
```diff
@@ -63,6 +63,8 @@ func bpfmanCreateNodeFirewallApplication(ctx context.Context, c client.Client, o
 			return fmt.Errorf("failed to get BpfApplication: %v for obj: %s", err, obj.Name)
 		}
 	} else {
+		// object exists repopulate it with the new configuration and update it
+		prepareBpfApplication(&bpfApp, obj, dbg)
```
Comment (anfredette) [inline pkg/bpf-mgr/lib-bpfman.go:67]: Should you move the prepareBpfApplication() call on line 53 to after line 57?  As-is, I think in the case where the object is found, the bpfApp created on line 53 is being overritten by the c.Get() call.

**Review by anfredette — CHANGES_REQUESTED** [review]
Good catch.  This makes sense based on the behavior we were seeing.  Other than the one comment on moving the first prepareBpfApplication(), this looks good.

File: pkg/bpf-mgr/lib-bpfman.go (hunk 1 above)
Comment (msherif1234) [reply pkg/bpf-mgr/lib-bpfman.go:67]: if Get fail it won't be overwritten I think, but maybe saver move inside

Comment (anfredette) [reply pkg/bpf-mgr/lib-bpfman.go:67]: I agree.  I think it works like it is, but you're calling it twice for the case where the get succeeds and I don't think you need to.

Comment (msherif1234) [reply pkg/bpf-mgr/lib-bpfman.go:75]: done

File: pkg/bpf-mgr/lib-bpfman.go [hunk 2]
```diff
@@ -145,9 +153,9 @@ func updateBpfApplication(ctx context.Context, c client.Client, bpfApp *bpfmanio
 	for _, p := range bpfApp.Spec.Programs {
 		switch p.Type {
 		case bpfmaniov1alpha1.ProgTypeXDP:
-			klog.Infof("Updating BpfApplicatopn for XDP Prog: %v for INFW obj: %v", *p.XDP, objName)
+			klog.Infof("Updating BpfApplication for XDP Prog: %v for INFW obj: %v", *p.XDP, objName)
 		case bpfmaniov1alpha1.ProgTypeTCX:
-			klog.Infof("Updating BpfApplicatopn for TCX Prog: %v for INFW obj: %v", *p.TCX, objName)
+			klog.Infof("Updating BpfApplicaton for TCX Prog: %v for INFW obj: %v", *p.TCX, objName)
```
Comment (anfredette) [inline pkg/bpf-mgr/lib-bpfman.go:158]: Just a nit:

```suggestion
			klog.Infof("Updating BpfApplication for TCX Prog: %v for INFW obj: %v", *p.TCX, objName)
```

**Review by anfredette — APPROVED** [review]
Just a nit on the log message.  Otherwise, LGTM.

**Comment by frobware** [comment]
I ran the test highlighted in the PR description - which passed.

LGTM

**Pull request merged**